Contributions/Using_Linux_in_a_Safe_System.md

```diff
@@ -0,0 +1,348 @@
+necessary to define what is required at system level in terms of both availability
+and integrity.
+
+The simpler - and most common - case is to require detection of interference
+affecting components that are recipient of safety requirements.
+
+Interference affecting a component with safety requirements must be both detected
+and reported, in a way that is compatible with the associated time requirements.
+At this point, appropriate action(s) will be taken, in accordance with the
+pre-established policy, to ensure adequate availability
+(e.g. restarting, onlining of backup system, etc.)
+
+As mentioned earlier, guaranteeing availability can be much more difficult
+and/or costly.
+
+* **Application Monitoring**
+
+  Monitoring is a simple approach to detection of interference that affects
+  component with safety requirement.
+  
+  It can save from extensive hardening.
+
+  However two conditions must be met:
+  1. It must be possible to express a simple check, in a way that can repeatedly,
+     reliably and unequivocally produce a YES/NO verdict about the health parameter
+     of interest
+  3. The observation whether the condition is met (or not) can be performed from a
+     context having same or better safety qualification.
+     And the context is able to take whatever action might be dictated by the
+     specific policy in place.
+
+  Assuming that the these two conditions are met (e.g. the system contains either a
+  watchdog or an equivalent component -- more on this later) then the opportunistic
+  approach is to identify an high-level activity that can be tied to the monitor.
+
+  A very crude example of this validation could be to require that application
+  responsible of performing safety relevant operations would ping a watchdog,
+  periodically.
+
+  As long as the application pings the watchdog within the set deadline, the watchdog
+  does not trigger any exceptional operation.
+
+  The example is crude, but it illustrates how - for example - all the internal linux
+  components involved with booting, loading and executing the application are not
+  subject to any safety requirement, because only the final product of their operation
+  is monitored.
+
+  In this case one must accept, though, that safety claims are possible only for
+  what is being monitored, either directly or indirectly.
+
+  If this is not acceptable, then additional requirements must be formulated,
+  and additional monitoring (or other means of mitigation) must be deployed.
+
+  With regard to the example, having a monitor that detects excessive skew in
+  the period used by the application to ping, will not ensure that other parts of
+  the application are free from interference and that they will function correctly.
+
+  Never the less, carefully choosing how/where to implement the monitor and how to
+  prime it can avoid expensive, tedious, error-prone involvement of large chunks
+  of the kernel architecture in supporting safety claims.
+
```

Comment:
The sentence above is a bit confusing IMO. Till now you have focused on interfernce, whereas now you seem to move also to other safety claims.
So for example the WTD mechanism can detect:
- any temporal failure mode deriving from interference from components outside those functionally needed to run the safety app
- any temporal failure mode deriving from failure modes of the components functionally supporting the safety app

In principle the failure modes above are different as the map to different safety claims....so what are we focusing on here ?